### ISSUE TITLE
Examples section and doctest output containing underlined text is being mistaken for unknown numpy docstring sections

### ISSUE THREAD

**Issue by oliversheridanmethven** [jsh9/pydoclint#300]
**The problem**

If the repr that gets printed that contains something looking like
```
Something
---------
```
then in the `Examples` section in the doctest, this repr output in the doctest get's falsely confused for an unknown numpy docstring section, 

**Expected Behaviour**

The parser used by pydoctest should be aware of the `<BLANKLINE>` syntax for doctest and the new lines that can separate doctests, and determine if this is part of the output a doctest is expecting, or a genuine error in the docstring.

**MWE**
I have a docstring and doctest that look like the following in the following MWE:
```python
"""A pretty object example"""

from __future__ import annotations

import textwrap


class PrettyObject:
    """A pretty object"""

    def __str__(self) -> str:
        """
        String representation of the object for str use.

        Returns
        -------
        str
            A pretty string representation of the object.
        """
        return repr(self)

    def __repr__(self) -> str:
        """
        String representation of the object for repr use.

        Returns
        -------
        str
            A pretty string representation of the object.
        """
        return textwrap.dedent(
            """
            First
            -----

            some info

            Second
            ------

            some more info
            that is also useful
            """
        )


def foo() -> PrettyObject:
    """
    Computes a pretty object.

    Returns
    -------
    PrettyObject
        A pretty object.

    Examples
    --------
    >>> foo()
    <BLANKLINE>
    First
    -----
    <BLANKLINE>
    some info
    <BLANKLINE>
    Second
    ------
    <BLANKLINE>
    some more info
    that is also useful
    <BLANKLINE>
    """
    return PrettyObject()
```

In this example, the doctest passes, but pydoclint complains with
```
DOC001 Function/method `foo`: Potential formatting errors in docstring. Error message: Unsupported numpy docstring sections: "First", "Second" (Note: DOC001 could trigger other unrelated violations under this function/method too. Please fix the docstring formatting first.)
```
